channels/audin/server/audin.c

```diff
@@ -559,8 +573,12 @@ static UINT audin_server_packet_send(audin_server_context* context, wStream* s)
 	WINPR_ASSERT(context);
 	WINPR_ASSERT(s);
 
-	if (!WTSVirtualChannelWrite(audin->audin_channel, (PCHAR)Stream_Buffer(s),
-	                            Stream_GetPosition(s), &written))
+	const size_t pos = Stream_GetPosition(s);
+	if (pos > UINT32_MAX)
+		return ERROR_INVALID_PARAMETER;
```

Comment:
This should probably output a warning message like with the other cases below (something like "Oversized DVC Data message").

Reply:
done. also for smartcard the same.